src/queries/dataRequests/user/communication-logs.sql

```diff
@@ -0,0 +1,268 @@
+      },
+      {
+        "columnName": "filenames",
+        "type": "string",
+        "nullable": true,
+        "description": "Names of any attached files related to the communication."
+      }
+    ]
+  },
+  "filters": [
+    {
+      "name": "regionIds",
+      "type": "integer[]",
+      "display": "Region IDs",
+      "description": "One or more values for 1 through 12 representing the region IDs."
+    },
+    {
+      "name": "communicationDate",
+      "type": "date[]",
+      "display": "Communication Date Range",
+      "description": "Two dates defining a range for the communicationDate. If only one date is supplied, the range is from the supplied date to the current timestamp. If no dates are supplied, this filter is ignored."
+    },
+    {
+      "name": "recipients",
+      "type": "string[]",
+      "display": "Recipient Names",
+      "description": "One or more recipient names to filter the results."
+    },
+    {
+      "name": "users",
+      "type": "string[]",
+      "display": "User Names",
+      "description": "One or more user names to filter the results."
+    },
+    {
+      "name": "role",
+      "type": "string[]",
+      "display": "Role Names",
+      "description": "One or more role names to filter the results."
+    },
+    {
+      "name": "method",
+      "type": "string[]",
+      "display": "Communication Methods",
+      "description": "One or more communication methods to filter the results, such as 'email' or 'phone'."
+    },
+    {
+      "name": "uei",
+      "type": "string[]",
+      "display": "UEI Values",
+      "description": "One or more UEI values to filter the results."
+    }
+  ],
+  "sorting": {
+    "default": [
+      { "level": 1, "name": "name", "order": "ASC" },
+      { "level": 2, "name": "communicationDate", "order": "ASC" }
+    ]
+  }
+}
+*/
+SELECT
+    r.name,
+    r.uei,
+    u.name AS "user",
+    STRING_AGG(DISTINCT rr.name, ' ') AS "roles",
+    cl."createdAt",
+    COALESCE(cl.data ->> 'method', '') AS "method",
+    cl.data ->> 'result' AS "result",
+    COALESCE(cl.data ->> 'purpose', '') AS "purpose",
+    COALESCE(cl.data ->> 'duration', '') AS "duration",
+    COALESCE(cl.data ->> 'regionId', '') AS "region",
+    CASE
+		WHEN data ->> 'communicationDate' ~ '^[0-9]{1,2}/[0-9]{1,2}/[0-9]{4}$' THEN TO_DATE(data ->> 'communicationDate', 'MM/DD/YYYY')
+		WHEN data ->> 'communicationDate' ~ '^[0-9]{1,2}/[0-9]{1,2}/[0-9]{2}$' THEN TO_DATE(data ->> 'communicationDate', 'MM/DD/YY')
+		WHEN data ->> 'communicationDate' ~ '^[0-9]{1,2}-[0-9]{1,2}-[0-9]{2}$' THEN TO_DATE(data ->> 'communicationDate', 'MM-DD-YY')
+		WHEN data ->> 'communicationDate' ~ '^[0-9]{1,2}/[0-9]{1,2}//[0-9]{2}$' THEN TO_DATE(regexp_replace(data ->> 'communicationDate', '//', '/'), 'MM/DD/YY')
+		WHEN data ->> 'communicationDate' ~ '^[0-9]{1,2}/[0-9]{1,2}/[0-9]{4}?[0-9]{1,2}.[0-9]{1,2}$' THEN TO_DATE(LEFT(data ->> 'communicationDate', 10), 'MM/DD/YYYY')
+		ELSE NULL
```

Comment:
It's a good thing we don't have that many comlogs because this is absurdly inefficient. We already chatted about the ticket for fixing this, correct?

Reply:
Data: https://jira.acf.gov/browse/TTAHUB-3451
frontend: https://jira.acf.gov/browse/TTAHUB-3452
backend: https://jira.acf.gov/browse/TTAHUB-3453